### PR TITLE
Add ListOptions for listing user migrations

### LINKED ISSUE #2416: List User Migrations: Add support for pagination
The [ListUserMigrations()](https://github.com/google/go-github/blob/3a432d62bf0318c14b2d1d7415bf94ea1a884342/github/migrations_user.go#L100) currently doesn't support pagination options.

We should add it.

PS: Will be creating a pull request shortly.

### PR DESCRIPTION
**Pull request by amitsaha**
Fix for https://github.com/google/go-github/issues/2416

### CODE REVIEW

**Comment by amitsaha** [comment]
> @amitsaha - it looks like one of the examples needs to be updated. Please run `go test ./...` from the base directory of the repo to make sure all tests pass. Please see CONTRIBUTING.md for more details. Thanks.

Thanks. I did that before I pushed my branch:

```
$ go test ./...
go: downloading golang.org/x/oauth2 v0.0.0-20180821212333-d2e6202438be
go: downloading golang.org/x/net v0.0.0-20210226172049-e18ecbb05110
ok  	github.com/google/go-github/v45/github	0.840s
?   	github.com/google/go-github/v45/test/fields	[no test files]
?   	github.com/google/go-github/v45/test/integration	[no test files]
```
From the linting failures above I can see that there is an example needed to be updated, so I can see the failure when I run `go test` from within the `example` directory.

Not sure if I am doing something wrong?

Updating with the fix.

**Review by gmlewis — APPROVED** [review]
Thank you, @amitsaha !
LGTM.

Awaiting second LGTM from any other contributor to this repo before merging.

**Comment by amitsaha** [comment]
Thanks @gmlewis !

File: github/migrations_user.go
Comment (raynigon) [inline github/migrations_user.go:105]: It would be nice if this case got tested.

**Review by raynigon — APPROVED** [review]
Can be merged, but it would be nice if the error case test gets added first.

**Comment by amitsaha** [comment]
@gmlewis I suppose you would want to me to add a test as suggested by @raynigon ?

File: github/migrations_user.go
Comment (amitsaha) [reply github/migrations_user.go:105]: @gmlewis @raynigon I am actually not sure how I can test this path..The `ListOptions{}` takes in `Page` and `PerPage`. Is there an example you can point me to?

Comment (gmlewis) [reply github/migrations_user.go:105]: Ah, right... with the fixed URL, I'm not sure this can be tested, so let's leave it as-is.  Thanks, @amitsaha .

**Comment by amitsaha** [comment]
@gmlewis Thanks! I know we had a new release just days back, is there any chance I can request you for a new release with this change?

**Comment by gmlewis** [comment]
> @gmlewis Thanks! I know we had a new release just days back, is there any chance I can request you for a new release with this change?

Yes, I think the worst thing that would happen is I might annoy some people, but that's OK.  I've done it before and survived. :joy:

**Comment by amitsaha** [comment]
> > @gmlewis Thanks! I know we had a new release just days back, is there any chance I can request you for a new release with this change?
> 
> Yes, I think the worst thing that would happen is I might annoy some people, but that's OK. I've done it before and survived. joy

Thanks a lot!

**Comment by gmlewis** [comment]
https://github.com/google/go-github/releases/tag/v47.0.0 now contains this change.

**Pull request merged**
